File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/RandomCategorizer.java
package cz.fb.manaus.reactor.categorizer;

import cz.fb.manaus.core.category.BetCoverage;
import cz.fb.manaus.core.category.categorizer.SettledBetCategorizer;
import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.time.Clock;
import java.util.Random;
import java.util.Set;

@Component
public class RandomCategorizer implements SettledBetCategorizer {

    private final Random random = new Random(Clock.systemUTC().millis());

    @Override
    public boolean isSimulationSupported() {
        return false;
    }

    @Override
    public Set<String> getCategories(SettledBet settledBet, BetCoverage coverage) {
        int randInt = random.nextInt(5);
        return Set.of("random_" + randInt);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/RandomCategorizer.kt
package cz.fb.manaus.reactor.categorizer

import cz.fb.manaus.core.category.categorizer.RealizedBetCategorizer
import cz.fb.manaus.core.model.RealizedBet
import org.springframework.stereotype.Component
import java.time.Clock
import java.util.*

@Component
class RandomCategorizer : RealizedBetCategorizer {

    private val random = Random(Clock.systemUTC().millis())

    override val isSimulationSupported: Boolean = false

    override fun getCategories(realizedBet: RealizedBet): Set<String> {
        val randInt = random.nextInt(5)
        return setOf("random_$randInt")
    }

}
